decimal.go

```diff
@@ -1307,6 +1307,31 @@ func (d NullDecimal) MarshalJSON() ([]byte, error) {
 	return d.Decimal.MarshalJSON()
 }
 
+// UnmarshalText implements the encoding.TextUnmarshaler interface for XML
+// deserialization
+func (d *NullDecimal) UnmarshalText(text []byte) error {
+	str := string(text)
+
+	if str == "" {
+		d.Valid = false
+		return nil
+	}
```

Comment:
Is this case necessary? Wouldn't the below if statement be enough?

Reply:
Decimal.Unmarshal returns error. For NullDecimal it is not an error as NullDecimal.Valid will be set to false.

Reply:
It is similar to how NullDecimal.UnmarshalJSON works.

Reply:
Ok, got it :3 Thanks!
I will get back to your PR tomorrow if you don't mind :) I would love to check the unit test too

Reply:
Any updates?

Reply:
I'm so sorry :<. I completely forgot about your pull request Bharath. I'm leaving in about an hour for holiday dinner with family, but I will check those unit test tomorrow.
I'm adding this to my TODO list, so I will not forget about it :D